packages/core/src/controllers/subscriber.ts

```diff
@@ -401,4 +410,16 @@ export class Subscriber extends ISubscriber {
       throw new Error(message);
     }
   }
+
+  private async restartToComplete() {
+    if (this.restartInProgress) return;
```

Comment:
Is there a missing negation `!` here or am I misunderstanding it? Seems like it returns immediately if a restart is in progress when the intention seems to be to create a locking mechanism to hold off on the `isInitialized` check being called if a restart is in progress.